File: thread/python/thread2.py
import threading
import time

class Mythread(threading.Thread):
    def __init__(self, name):
        threading.Thread.__init__(self)
        self.name = name  
        
    def run(self):
        print("{} start ".format (threading.current_thread().name))
        time.sleep(3)
        print("{} end ".format (threading.current_thread().name))

def main():
    for i in range(5):
        t = Mythread("thread {}".format(i))              
        t.start()     
        
if __name__ == "__main__":
    main()                

File: thread/java/thread2.java
class MyThread1 extends Thread {
   public MyThread1(){
        setName("MyThread1");
    }
    @Override
    public void run() {
        for (int i = 0; i < 5; i++) {
            System.out.println(getName()+" : " + i);
            try {   
                Thread.sleep(1000);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }
}
class MyThread2 extends Thread {
    public MyThread2(){
        setName("MyThread2");
    }
    @Override
    public void run() {
        for (int i = 0; i < 5; i++) {
            System.out.println(getName()+" : " + i);
            try {   
                Thread.sleep(1000);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }
}
public class thread2 {
    public static void main(String[] args) {
        Thread thread1 = new MyThread1();
        Thread thread2 = new MyThread2();
        thread1.start();
        thread2.start();
    }
}